Monika After Story/game/event-handler.rpy

```diff
@@ -1459,6 +1461,112 @@ init -1 python in evhand:
         )
     )
 
+
+    class EventListItem(object):
+        """
+        Representation of an EventListItem (ELI)
+        """
+        # default values for each main param
+        # make sure this matches push/queue
+        DEFAULT_VALUES = (
+            False, # notify
+            None, # context
+        )
+
+        ITEM_LEN = len(DEFAULT_VALUES) + 1 # defaults + event label
+
+        IDX_EVENT_LABEL = 0
+        IDX_NOTIFY = 1
+        IDX_CONTEXT = 2
+
+        def __init__(self, data):
+            """
+            Constructor
+
+            IN:
+                data - the data directly from event list
+            """
+            self._eli = data
+
+        @staticmethod
+        def build(evl, *args):
+            """
+            Builds an ELI.
+
+            IN:
+                evl - event label
+                *args - the other args for an EventListItem.
+
+            RETURNS: EventListItem object
+            """
+            return EventListItem(EventListItem._build_raw(evl, *args))
+
+        @staticmethod
+        def _build_raw(evl, *args):
```

Comment:
so when I started writing this class, I thought about what ways I would need to create this object for:
1. use an item directly from `event_list`
2. create one from named params
3. create one using older data - for upgrading users

option 2 is the straightforward one - use named params for each thing, so I started with that for the constructor.
option 1 would be a static method that takes the item and splits it into args for the constructor - modify this a little bit and now we can handle option 3 with some safety checks (this would also need constants for default values)

this was fine until I thought about what happens when a new data element is added:
1. add the param to push/queue
2. add the param to the constructor
3. add handling for the param in the option 1/3 static method
4. add a default value entry
5. add a private var for the param
6. add the get function for the param

Steps 2, 3, and 5 are kind of annoying - its basically repeated param/documentation from step 1. So how can this be removed? 

Generic kwargs gets rid of step 2 and 5, but still need to do "arg name", which is same concept as the item indices. Step 3 still needs manual changes because the ordered data (`event_list`) would still need to be split and properly passed into kwargs.

Generic `*args` means that we only need to consider data position, which is guaranteed since we would never change the order in `event_list` - and this is a hard never, like `Event` itself. This also makes the static method covering option 1/3 even easier to write - just use list splicing to make up the difference when using old data (list splicing even handles if the the splice index is larger than the actual list, i.e: `abc = [1,2,3] -> abc[10:] -> []`). Now we end up with these steps when a new param is added:
1. add the param to push/queue
2. add a default value entry
3. add the index constant (honestly not even needed but I added for the sake of magic numbers)
4. add the get function for the param

Way simpler to add more. Is it dangerous for callers? You bet! but it's only usage outside of `MASEventList` should be read-only data retrieval. If I called this `MASEventListItem` then I would have made it safer.

Regarding your points in the other comment:
1. Mentioned above that order will not change in `event_list`, so the order flexibility is not needed. But okay, let's imagine that it does change - well the `_raw` function means that we can change the order when the EventListItem is actually added to the `event_list`. We can also change the indices and default values order to handle input. I think even a named args solution would require us to make a change in the output to support a new order, so we save maybe 1 set of changes for something that rarely will happen?
2. Agreed, but with list splicing we can make args safe, too.
3. We'd still have to keep the names somewhere, so this seems more like a 1:1 replacement. (we all seem to have varying opinions on magic strings, too)
4. mentioned above that kwargs 1 step extra for each new data element compared to args. also mentioned above that the "arg name" is same conceptually as the unique id
5. Agreed, but deprecations will be almost never. If a deprecation ever occurs, it would become an unused data element, similar to what happens with `Event`. Wasteful, yes, but negligible data usage.

Reply:
Even if this API isn't public, I still don't feel safe knowing we rely on precise arg position. I'd rather be extra safe now than sorry later when we do something dum and save invalid data in persistent. List slicing doesn't really make it much better to me.

It's harder to read what's going on in `data = list((evl, ) + args + EventListItem.DEFAULT_VALUES[len(args):])`. I can understand it, it's just not as clear to read.

I think keeping default values in a `dict` is better. Yes, it's still a value + key, but it feel more cohesive because everything is together. If we could use enums, I'd see idx work, but right now `IDX_EVENT_LABEL` is just an `int` and doesn't tell much when working with the code.

I won't push this further tho and leave the final decision to you.

Reply:
>I still don't feel safe knowing we rely on precise arg position. 

I mainly feel fine with relying on arg position because `event_list` is structured on position, so even if some of the `EventListItem` functions don't rely on position, positioning must be relied on somewhere in there for the thing to work. If `event_list` tuples were changed to dicts, then I think it would be reasonable for `EventListItem` to use more named-based/dict-based handling (and I don't think there are compelling reasons to change away from tuples).

>final decision 
You made great points overall, and I agree that my solution is not as robust in comparison, but I still think this implementation is sufficient for the current use case. If the use case changes (mainly adding dozens of properties to event list items or making event list item manipulation a norm), then we can revisit this - this is luckily abstracted enough that if it changes again, it wouldn't affect public API usage anywhere.